src/uct/api/uct_def.h

```diff
@@ -405,4 +406,62 @@ typedef ucs_status_t (*uct_tag_unexp_rndv_cb_t)(void *arg, unsigned flags,
                                                 const void *rkey_buf);
 
 
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Tuning parameters for an UCT endpoint created by @ref uct_ep_create.
+ */
+typedef struct uct_ep_params {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_ep_params_field. Fields not specified in this mask
+     * would be ignored. Provides ABI compatibility with respect to adding new
+     * fields.
+     */
+    uint64_t                          field_mask;
+    /**
+     * Interface to create the endpoint on. This is a mandatory field.
+     */
+    uct_iface_h                       iface;
+    /**
+     * User data associated with the endpoint.
+     */
+    void                              *user_data;
+    /**
+     * The device address to connect to on the remote peer. Should be defined
+     * together with @ref uct_ep_params_t::iface_addr to create an endpoint
+     * connected to a remote interface.
+     */
+    const uct_device_addr_t           *dev_addr;
+     /**
+      * Create an endpoint which is connected to remote interface.
+      * The interface address to connect to on the remote peer.
+      * @note requires @ref UCT_IFACE_FLAG_CONNECT_TO_IFACE capability.
+      */
+    const uct_iface_addr_t            *iface_addr;
+    /**
+     * The sockaddr to connect to on the remote peer. If set, @ref uct_ep_create
+     * will create an endpoint for a connection to the remote peer, specified by
+     * its socket address.
+     * @note The interface in this routine requires the
+     * @ref UCT_IFACE_FLAG_CONNECT_TO_SOCKADDR capability.
+     */
+    const ucs_sock_addr_t             *sockaddr;
+    /**
+     * Required @ref uct_cb_flags to indicate where callbacks can be invoked
+     * from.
```

Comment:
It's not always required, right? only for sockaddr mode?
what does it mean "from"? need to mention thread-safety